system/src/system_connection_manager.cpp

```diff
@@ -0,0 +1,489 @@
+        return SYSTEM_ERROR_NO_MEMORY;
+    }
+
+    metrics->txBuffer = txBuffer;
+    metrics->rxBuffer = rxBuffer;
+    return 0;
+};
+
+int ConnectionTester::sendTestPacket(ConnectionMetrics* metrics, int length) {
+    int r = 0;
+    // Only send a new packet if we have received the previous one
+    if (metrics->txPacketCount == metrics->rxPacketCount) {
+        CHECK(generateTestPacket(metrics, length));
+
+        int r = sock_send(metrics->socketDescriptor, metrics->txBuffer, metrics->testPacketSize, 0);
+        if (r > 0) {
+            metrics->txPacketStartMillis = millis();
+            metrics->txPacketCount++;
+            metrics->testPacketSequenceNumber++;
+            metrics->txBytes += metrics->testPacketSize;
+        } else {
+            LOG(ERROR, "test sock_send failed %d errno %d interface %d", r, errno, metrics->interface);
+            return SYSTEM_ERROR_NETWORK;
+        }
+        
+        LOG(TRACE, "sock %d packet # %d tx > %d", metrics->socketDescriptor, metrics->txPacketCount, r);
+    }
+    return r;
+};
+
+int ConnectionTester::receiveTestPacket(ConnectionMetrics* metrics) {
+    int r = sock_recv(metrics->socketDescriptor, metrics->rxBuffer, metrics->testPacketSize, MSG_DONTWAIT);
+    if (r > 0) {
+        CHECK_TRUE(r == metrics->testPacketSize, SYSTEM_ERROR_BAD_DATA);
+
+        if (memcmp(metrics->rxBuffer, metrics->txBuffer, r)) {
+            // Did not receive the exact same message sent
+            LOG(ERROR, "test socket on interface %d did not receive the same echo data");
+            return SYSTEM_ERROR_BAD_DATA;
+        }
+
+        metrics->totalPacketWaitMillis += (millis() - metrics->txPacketStartMillis);
+        metrics->rxPacketCount++;
+        metrics->rxBytes += metrics->testPacketSize;
+    } else {
+        LOG(ERROR, "test sock_recv failed %d errno %d interface %d", r, errno, metrics->interface);
+        return SYSTEM_ERROR_NETWORK;
+    }
+    
+    LOG(TRACE, "sock %d packet # %d rx < %d", metrics->socketDescriptor, metrics->rxPacketCount, r);
+    return r;
+};
+
+int ConnectionTester::generateTestPacket(ConnectionMetrics* metrics, int packetDataLength) {
+    auto network = spark::Network.from(metrics->interface);
+    if (!network) {
+        LOG(ERROR, "No Network associated with interface %d", metrics->interface);
+        return SYSTEM_ERROR_BAD_DATA;
+    }
+
+    DTLSPlaintext_t msg = {
+        REACHABILITY_TEST_MSG, // DTLS Message Type
+        {0xfe, 0xfd},          // DTLS type 1.2
+        0x8000,                // Differentiate interfaces by epoch field
+        {},                    // Sequence number
+        0                      // Payload length
+    };
+
+    int headerLength = sizeof(msg);
+    int totalMessageLength = packetDataLength + headerLength;
+    //LOG(TRACE, "packetDataLength %u headerLength %d totalMessageLength %d", packetDataLength, headerLength, totalMessageLength);
+
+    msg.epoch |= metrics->interface;
```

Comment:
Nit/TODO: use the same 'fixed' `network_interface_index` universally on all platforms, same as in vitals/diagnostics.

Reply:
Should be resolved now